cmake/examples.cmake

```diff
@@ -0,0 +1,28 @@
+function( add_example ARG_NAME )
+	list( APPEND ARG_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}/examples/${ARG_NAME}" )
+	set( SOURCES "" )
+	set( RESOURCES "" )
+	foreach( DIR ${ARG_DIRECTORIES} )
+		file( GLOB GLOB_SOURCES ${DIR}/*.c ${DIR}/*.h )
+		list( APPEND SOURCES ${GLOB_SOURCES} )
+		file( GLOB GLOB_RESOURCES ${DIR}/*.rc )
+		list( APPEND RESOURCES ${GLOB_RESOURCES} )
+	endforeach()
+	add_executable( ${ARG_NAME} ${SOURCES} )
+	target_link_libraries( ${ARG_NAME} PUBLIC adikted )
+	find_package( SDL REQUIRED )
+	target_link_libraries( ${ARG_NAME} PUBLIC ${SDL_LIBRARY} )
+	target_compile_definitions( ${ARG_NAME} PUBLIC ENTRY_CONFIG_USE_SDL )
+	target_link_libraries( ${ARG_NAME} PUBLIC X11 Math::Math )
```

Comment:
Any reason for doing `Math::Math` instead of just `m`? Seems to work find just using `m` - I made a pull request to your repo with the change.

Reply:
You need to install libadikted first and  run
```c
export PKG_CONFIG_PATH=$PREFIX/${libdir}/pkgconfig && ldconfig
```
I apologize for reopening the pull request.

Reply:
Alright, fixing mapslang compile issue can wait I'd rather have this PR merged sooner rather than later and invoking cmake from the root directory works just fine for now. I'll do a proper review when I get home from work tomorrow.

Reply:
I propose a package interface named DKFans::ADiKtED and resolve the dependencies for parent directories too.
Possible non -lm math library linker flags exist too.